homework4/task1.py

```diff
@@ -0,0 +1,10 @@
+import re
+
+string = "name=Amanda=sssss&age=32&&salary=1500&currency=euro"
+
+pattern = re.compile(r'([^=&]+)=([^&^s{6}(?==)]*)')
```

Comment:
r'([^=&]+)=([^&^s{6}(?==)]*)' 

Another = is not required after ?. You have ?== but ?= also works 
r'([^=&]+)=([^&^s{6}(?=)]*)'